source/dem/non_uniform_insertion.cc

```diff
@@ -122,82 +163,65 @@ NonUniformInsertion<dim>::create_random_number_container(
-    Utilities::MPI::this_mpi_process(communicator);
-
-  // Creating a particle counter
-  unsigned int particle_counter = 0;
-
-  for (unsigned int i = 0; i < this->number_of_particles_x_direction; ++i)
-    for (unsigned int j = 0; j < this->number_of_particles_y_direction; ++j)
-      {
-        // Adapt the last index to the dimensionality of the problem
-        unsigned int dim_nz =
-          (dim == 3) ? this->number_of_particles_z_direction : 1;
-        for (unsigned int k = 0; k < dim_nz; ++k)
-          {
-            if (this->particle_on_processor(i,
-                                            j,
-                                            k,
-                                            this_mpi_process,
-                                            Utilities::MPI::n_mpi_processes(
-                                              communicator)))
-              {
-                // We need to check if the number of inserted particles so far
-                // at this step (particle_counter) reached the total desired
-                // number of inserted particles at this step
-
-                if (particle_counter < this->inserted_this_step_this_proc)
-                  {
-                    Point<dim> position;
-                    // Obtaning position of the inserted particle
-                    // In non-uniform insertion, random numbers were created and
-                    // are added to the position of particles. In order to
-                    // create more randomness, the random vector is read once
-                    // from the beginning and once from the end to be used in
-                    // positions [0] and [1]
-                    position[0] =
-                      insertion_information.x_min +
-                      ((i + 0.5) * insertion_information.distance_threshold -
-                       random_number_vector[particle_counter]) *
-                        this->maximum_diameter;
-                    position[1] =
-                      insertion_information.y_min +
-                      ((j + 0.5) * insertion_information.distance_threshold -
-                       random_number_vector[this->inserted_this_step_this_proc -
-                                            particle_counter - 1]) *
-                        this->maximum_diameter;
-                    if (dim == 3)
-                      {
-                        position[2] =
-                          insertion_information.z_min +
-                          ((k + 0.5) *
-                             insertion_information.distance_threshold -
-                           random_number_vector[particle_counter]) *
-                            this->maximum_diameter;
-                      }
-
-                    insertion_positions.push_back(position);
-                    particle_counter++;
-                  }
-              }
-          }
-      }
+  std::vector<int> insertion_index;
+  insertion_index.reserve(3);
```

Comment:
```suggestion
  insertion_index.resize(3);
```
This should be a resize here because you want to contain 3 elements and you are not pushing back. Reserve is only when you want to pre-allocate without changing size.

https://stackoverflow.com/questions/13029299/stdvectorresize-vs-stdvectorreserve

Reply:
corrected. thank you for the tip.